src/components/publishing/imageset_preview.tsx

```diff
@@ -1,105 +1,131 @@
+  ${Fonts.unica("s14", "medium")}
+`
+const SubTitleCount = styled.div`
+  ${Fonts.unica("s14")}
+  margin-left: 20px;
 `
 const IconContainer = styled.div`
-  width: 32px;
-  margin-bottom: 10px;
+  height: 45px;
   position: relative;
+  margin-left: 40px;
+  text-align: right;
+  > svg {
+    height: 98%;
+  }
 `
 
 class ImagesetPreview extends Component<any, any> {
-  constructor(props) {
-    super(props)
-    this.state = {
-      visibleImages: this.getVisibleImages(this.props.images),
+  getImageUrl() {
+    const image = this.props.section.images[0]
+    const src = image.url ? image.url : image.image
+    return src
+  }
+  textSection(layout) {
+    if (layout === "mini") {
+      return (
+        <MiniWrapper>
+          {this.image(layout)}
+          <MiniInner>
+            {this.title()}
+            {this.icon()}
+          </MiniInner>
+        </MiniWrapper>
+      )
+    } else {
+      return (
+        <FullWrapper>
+          {this.title()}
+          {this.icon()}
+        </FullWrapper>
+      )
     }
   }
-
-  getVisibleImages(images) {
-    let widths = []
-    let hidden = 0
-    images.map((item, i) => {
-      const adjustedWidth = 150 * item.width / item.height
-      widths.push(adjustedWidth)
-      const total = widths.reduce((a, b) => a + b, 0)
-      const margins = widths.length * 10
-      if (total + margins + 50 > 560) {
-        hidden = hidden + 1
-      }
-    })
-    return widths.length - hidden
+  image(layout) {
+    const src = this.getImageUrl()
+    const width = layout === "full" ? "100%" : "auto"
+    const height = layout === "mini" ? "100%" : "auto"
+    return <img src={src} width={width} height={height} />
   }
-
-  renderImages(images) {
-    const items = images.slice(0, 4).map((item, i) => {
-      const src = item.image ? item.image : item.url
-      if (i < this.state.visibleImages) {
-        return (
-          <img
-            key={"imageset-" + i}
-            src={src || ""}
-            height="150"
-            className="imageset-preview__image"
-            style={{ marginRight: 10 }}
-          />
-        )
-      }
-    }, this)
-    return items
+  title() {
+    return (
+      <TitleWrapper data-layout={this.props.section.layout}>
+        <Title>{this.props.section.title}</Title>
+        <SubTitle>
+          <SubTitlePrompt>View Slideshow</SubTitlePrompt>
+          <SubTitleCount>{this.props.section.images.length} Images</SubTitleCount>
+        </SubTitle>
+      </TitleWrapper>
+    )
+  }
+  icon() {
+    return (
+      <IconContainer>
+        <IconImageset />
+      </IconContainer>
+    )
   }
-
   render() {
-    const { images } = this.props
-    const length = {
-      position: "absolute",
-      left: 8,
-      top: 14,
-    }
-    if (images.length > 9) {
-      length.left = 4
-    }
-
+    const { layout } = this.props.section
+    const image = layout === "full" ? <img src={this.getImageUrl()} width="100%" /> : ""
```

Comment:
Do you really need an empty string if `layout === "mini`? If not, the code would be more portable to React Native if you’d use `null` instead. (React Native does not accept an empty string without explicit `Text` tag.)